src/mock.ts

```diff
@@ -91,57 +101,68 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
   // and completes the customMock object with any fields
   // defined on genericMock
   // only merges objects or arrays. Scalars are returned as is
-  function mergeMocks(genericMockFunction, customMock) {
+  function mergeMocks(genericMockFunction: IMockFn, customMock: any): any {
     if (Array.isArray(customMock)) {
-      return customMock.map((el) => mergeMocks(genericMockFunction, el));
+      return customMock.map((el: any) => mergeMocks(genericMockFunction, el));
     }
     if (isObject(customMock)) {
       return mergeObjects(genericMockFunction(), customMock);
     }
     return customMock;
   }
 
-  function assignResolveType(type) {
+  function getResolveType(namedFieldType: GraphQLNamedType) {
+    if ( (namedFieldType instanceof GraphQLInterfaceType) ||
+         (namedFieldType instanceof GraphQLUnionType) ) {
+        return namedFieldType.resolveType;
+    } else {
+        return undefined;
+    }
+  }
+
+  function assignResolveType(type: GraphQLType) {
     const fieldType = getNullableType(type);
     const namedFieldType = getNamedType(fieldType);
 
-    const oldResolveType = namedFieldType.resolveType;
+    const oldResolveType = getResolveType(namedFieldType);
     if (preserveResolvers && oldResolveType && oldResolveType.length) {
       return;
     }
 
     if (namedFieldType instanceof GraphQLUnionType ||
-        namedFieldType instanceof GraphQLInterfaceType
-    ) {
+        namedFieldType instanceof GraphQLInterfaceType) {
       // the default `resolveType` always returns null. We add a fallback
       // resolution that works with how unions and interface are mocked
-      namedFieldType.resolveType = (data, context, info) =>
-        info.schema.getType(data.typename);
+      namedFieldType.resolveType = (data: any, context: any, info?: GraphQLResolveInfo) => {
+        if ( undefined === info ) {
+            throw new Error(`no info provided for resolveType`);
+        }
+        return info.schema.getType(data.typename) as GraphQLObjectType;
+      };
     }
   }
 
-  const mockType = function mockType(type, typeName, fieldName) {
+  const mockType = function mockType(type: GraphQLType, typeName?: string, fieldName?: string): IResolveFn {
```

Comment:
```typescript
      // TODO mocking Interface and Union types will require determining the
      // resolve type before passing it on.
      // XXX we recommend a generic way for resolve type here, which is defining
      // typename on the object.
      if (fieldType instanceof GraphQLUnionType) {
        const randomType = getRandomElement(fieldType.getTypes());
        return Object.assign({ typename: randomType }, mockType(randomType)(root, args, context, info));
      }
      if (fieldType instanceof GraphQLInterfaceType) {
        const possibleTypes = schema.getPossibleTypes(fieldType);
        const randomType = getRandomElement(possibleTypes);
        return Object.assign({ typename: randomType }, mockType(randomType)(root, args, context, info));
      }
```

note that it's being called (recursion) without supplying those arguments,
so is it a bug?